src/Tests/dotnet.Tests/CommandTests/ToolInstallGlobalOrToolPathCommandTests.cs

```diff
@@ -351,6 +352,31 @@ public void WhenRunWithValidVersionRangeItShouldSucceed()
                     PackageVersion).Green());
         }
 
+        [Fact(Skip = "https://github.com/dotnet/sdk/pull/36021")]
+        public void WhenRunWithValidUnlistedVersionRangeItShouldSucceed()
```

Comment:
I think this test should be enabled in CI if it can work there.

Can you also add a test that checks that an unlisted package won't be installed if you don't specify a version?

Reply:
Thank you! A new test is added for unlisted package without version and the previous test was updated to use `DotnetCommand`